### PR TITLE
Move TestWorkflowRule to a dedicated module

### PR DESCRIPTION
**Pull request by GreyTeardrop**
## Overview

Decouple generic Temporal testing infrastructure and JUnit4-specific `TestWorkflowRule`. Move `TestWorkflowRule` to a dedicated `temporal-testing-junit4` module.

This opens up an opportunity for alternative test framework integrations, e.g. JUnit 5, Spock, Spek, etc.

### CODE REVIEW

File: temporal-testing-junit4/build.gradle [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+                }
+                usage('java-runtime') {
+                    fromResolutionResult()
+                }
+            }
+            pom {
+                name = 'Temporal Java SDK testing'
+                packaging = 'jar'
+                // optionally artifactId can be defined here
+                description = 'Temporal Workflow Java SDK testing'
```
Comment (mfateev) [inline temporal-testing-junit4/build.gradle:110]: Temporal Workflow Java SDK testing, JUnit 4.x integration

File: temporal-testing-junit4/build.gradle [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    withSourcesJar()
+}
+
+dependencies {
+    errorproneJavac('com.google.errorprone:javac:9+181-r4173-1')
+    errorprone('com.google.errorprone:error_prone_core:2.4.0')
+
+    api project(':temporal-testing')
+
+    implementation group: 'junit', name: 'junit', version: '4.13.1'
```
Comment (mfateev) [inline temporal-testing-junit4/build.gradle:43]: I think it should be api level dependency:

api group: 'junit', name: 'junit', version: '4.13.1'

**Review by mfateev — COMMENTED** [review]
Thanks a lot for the contribution!
Agree that removing JUnit dependency from the testing module is the right move.

File: temporal-testing-junit4/build.gradle (hunk 2 above)
Comment (GreyTeardrop) [reply temporal-testing-junit4/build.gradle]: You're right, I've changed the level to `api`.

File: temporal-testing-junit4/build.gradle (hunk 1 above)
Comment (GreyTeardrop) [reply temporal-testing-junit4/build.gradle]: Updated. I guess the `name` has to be updated as well, I've changed it to `Temporal Java SDK testing for JUnit 4`, not sure if that might be too long for a name though.

File: temporal-testing-junit4/build.gradle [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+            }
+        }
+
+    }
+
+    signing {
+        sign publishing.publications.mavenJava
+    }
+
+    // Uncomment to test local publishing and comment nexusPublishing
```
Comment (vitarb) [inline temporal-testing-junit4/build.gradle:147]: Nit. Let's remove this as we an simply run `/gradlew publishToMavenLocal` without having to change anything.

**Review by vitarb — APPROVED** [review]

**Pull request merged**